gitutil.py

```diff
@@ -0,0 +1,114 @@
+import os
+import subprocess
+
+import git
+
+from util import ensure_not_empty, ensure_directory_exists
+
+def update_submodule(
+    repo_path: str,
+    tree_ish: str,
+    submodule_path: str,
+    commit_hash: str,
+    author: str,
+    email: str,
+):
+    '''Update the submodule of a git-repository to a specific commit.
+
+    Create a new commit, with the passed tree-ish as parent, in the given repository.
+
+    Parameters
+    ------
+    repo_path : str
+        Path to a directory containing an intialised git-repo with a submodule to update.
+    tree_ish : str
+        Valid tree-ish to use as base for creating the new commit. This will be used as parent for the commit created by this function.
+        Example: 'master' for the head of the master-branch.
+    submodule_path : str
+        Path (relative to `repo_path`) to the submodule. Must be immediately below the root (i.e. `repo_path`) of the repository.
+    commit_hash : str
+        The hash the submodule should point to in the created commit. This should be a valid commit hash in the submodule's repository.
+    author : str,
+        Will be set as author of the created commit
+    email : str
+        Will be set for the author of the created commit
+
+    Returns
+    ------
+    str
+        The hexadecimal SHA-1 hash of the created commit
+    '''
+    ensure_directory_exists(repo_path)
+    repo_path = os.path.abspath(repo_path)
+
+    ensure_not_empty(tree_ish)
+    ensure_not_empty(submodule_path)
+    ensure_directory_exists(os.path.join(repo_path, submodule_path))
+    ensure_not_empty(commit_hash)
+    ensure_not_empty(author)
+    ensure_not_empty(email)
+
+    repo = git.Repo(repo_path)
+
+    # Fetch tree-ish from git.repo and create mktree-parseable string-representation of it.
+    # Since submodule-objects contain the SHA of the to-be-checked-out version of the
+    # target repository, we replace that during the string building.
+    tree = repo.tree(tree_ish)
+    tree_representation = '\n'.join(_tree_string_generator(tree, submodule_path, commit_hash))
+
+    # Pass the serialised trees to git mk-tree using GitPython. We cannot do this in GitPython
```

Comment:
isn't it just _one_ tree (which in turn contains a sequence of trees, blobs and submodule(s))?